benchmark_runner/common/clouds/BareMetal/bare_metal_operations.py

```diff
@@ -0,0 +1,284 @@
+                                            file_name=file_name,
+                                            parameter='openshift_version:',
+                                            value=f'"{openshift_version_data[0]}.{openshift_version_data[1]}"')
+
+    def update_provision_config(self):
+        """
+        This method update provision config file
+        :return:
+        """
+        # update ssh config - must add it and also mount it
+        self._ssh.run(f"echo Host provision > /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'HostName {self._provision_ip} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'User {self._user} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'IdentityFile {self._container_private_key_path} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'StrictHostKeyChecking no >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveInterval 30 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveCountMax 5 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"chmod 600 /{self._user}/.ssh/config")
+
+    @logger_time_stamp
+    def run_ocp_installer(self):
+        """
+        This method run ocp assisted installer
+        :return: True if installation success and raise exception if installation failed
+        """
+        """
+        This method run ocp assisted installer
+        :return: True if installation success and raise exception if installation failed
+        """
+        self.update_provision_config()
+        logger.info(f'Starting OCP assisted installer, Start time: {datetime.now().strftime(datetime_format)}')
+        # Must add -t otherwise remote ssh of ansible will not end
+        self._ssh.run(f"ssh -t provision \"{self._install_ocp_cmd()}\" ")
+        logger.info(f'OpenShift cluster {self._get_installation_version()} version is installed successfully, End time: {datetime.now().strftime(datetime_format)}')
+
+    @logger_time_stamp
+    def verify_install_complete(self):
+        """
+        This method verify that installation complete
+        :return: True if installation success and raise exception if installation failed
+        """
+        complete = self._wait_for_install_complete()
+        if not complete:
+            raise Exception(f'Installation failed')
+        else:
+            return True
+
+    @logger_time_stamp
+    def oc_login(self):
+        """
+        This method login to the cluster with new credentials
+        :return:
+        """
+        oc = OC(kubeadmin_password=self._get_kubeadmin_password())
+        oc.login()
+        return oc
+
+    @logger_time_stamp
+    def verify_cluster_is_up(self, oc: OC):
+        """
+        This method verify that master/worker nodes are up and running
+        :return:
+        """
+        master_nodes = oc.get_master_nodes()
+        # master always are 3
+        if len(master_nodes.split()) == 3:
```

Comment:
If you install on an SNO cluster, you'll only have 1.

Reply:
Fixed, add dedicate cluster_type environment variable to support SNO

Reply:
Added